code/clue.js

```diff
@@ -1,59 +1,238 @@
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age} years young`
+  document.getElementById('killerOccupation').innerHTML = mystery.killer.occupation
+  document.getElementById('killerDescription').innerHTML = mystery.killer.description
+  document.getElementById('killerImage').src = mystery.killer.image
+}
+document.getElementById('killerCard').onclick = pickKiller
+
+// FUNCTIONS pickWeapon 
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponName').innerHTML = mystery.weapon.name
+  document.getElementById('weaponFact').innerHTML = `${mystery.weapon.fact}`
+  document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight} lbs`
+
 }
+document.getElementById('weaponCard').onclick = pickWeapon
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//favouriteWeapon
+const favouriteWeapon = () => {
+  mystery.favouriteWeapon = randomSelector(weapons)
+}
+const shuffleFavouriteWeapon = () => {
+  mystery.favouriteWeapon = randomSelector(weapons)
+  document.getElementById('favouriteWeapon').innerHTML
+}
+shuffleFavouriteWeapon()
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+//This block of code is the one I really wanted to make work but somehow it never did:
+
+// const favourite = suspects.favouriteWeapon
+// const shuffleFavouriteWeaponDuplicate = new Array(6).fill(favourite)
+// const randomArray = [...weapons, ...favouriteWeaponDuplicate]
+// const pickNotSoRandomWeapon = () => {
+//   mystery.favouriteWeapon = randomArray[Math.floor(Math.random() * randomArray.length)]
+// }
+// pickNotSoRandomWeapon()
+
+// pickRoom
+const pickRoom = () => {
+  mystery.rooms = randomSelector(rooms)
+  document.getElementById('roomName').innerHTML = `${mystery.rooms}`
+}
+document.getElementById('roomCard').onclick = pickRoom
+
+// FUNCTION revealMystery that will be invoked when you click that button
+const thisMystery = document.getElementById('mystery')
+
+const revealMystery = () => {
+  if (mystery.killer == '' || mystery.weapon == '' || mystery.room == '') {
+    //   if (mystery.killer == null || mystery.weapon == null || mystery.room == null) -this prints no mystery to be revealed yet but it prints the same when all cards chosen, too
+    document.getElementById('mystery').innerHTML = `No mystery to be revealed yet!`
+  } else {
+    thisMystery.innerHTML = `The murder was committed by the ${mystery.killer.occupation} ${mystery.killer.firstName} ${mystery.killer.lastName} , room: ${mystery.rooms}, executed with a ${mystery.weapon.weight} lbs ${mystery.weapon.name}. ${mystery.weapon.fact}! This suspect's favourite weapon is: ${mystery.favouriteWeapon.name}. Spooky.`
+
+  }
+}
```

Comment:
I can see that the else message is not printed in the HTML. Not sure why the if conditional using null is not working for you, my function looks the same and works fine.

A tip is to check for possible errors using the console. When I typed mystery in the console (without clicking on any cards) for your page, I got this result: 

`{killer: null, weapon: null, room: null, favouriteWeapon: {…}}
`

Maybe it has something to do with the favorite weapon (or not lol)? Just speculating but I hope it can give you clues.  🤗

Also, in line 227 you declaring the variable for the mystery element id in html, I would suggest to use `const thisMystery `in line 232 as well.